doc/modules/linear_model.rst

```diff
@@ -897,6 +897,123 @@ to warm-starting (see :term:`Glossary <warm_start>`).
     .. [9] `"Performance Evaluation of Lbfgs vs other solvers"
             <http://www.fuzihao.org/blog/2016/01/16/Comparison-of-Gradient-Descent-Stochastic-Gradient-Descent-and-L-BFGS/>`_
 
+.. _Generalized_linear_regression:
+
+Generalized Linear Regression
+=============================
+
+Generalized Linear Models (GLM) extend linear models in two ways
+[10]_. First, the predicted values :math:`\hat{y}` are linked to a linear
+combination of the input variables :math:`X` via an inverse link function
+:math:`h` as
```

Comment:
First, this is standard literature nomenclature: link function g: g(E[Y]) = Xw and inverse link function h: h(Xw) = g^{-1}(Xw) = E[Y].

Second, we had this info/equation in the user guide, but some reviewer wanted it gone :cry:

Reply:
See here: https://78243-843222-gh.circle-artifacts.com/0/doc/modules/linear_model.html#id34

Reply:
I agree the link function `g` could be introduced e.g. below in the usage part.

I don't like using E[Y] because:

- we don't talk about E[Y] for the rest of the linear models, so why here
- it's actually E[Y/X], right? It seems to me that calling that E[Y] is another implicit nomenclature of GLMs, which we want to avoid in the UG precisely because it's implicit.

Reply:
> this is standard literature nomenclature

We should go with it then :) (my remarks are from somebody which is not familiar with GLMs)

> Second, we had this info/equation in the user guide, but some reviewer wanted it gone

I was not thinking to go until this level of detail but more about defining with a couple of words what is the link function. Regarding the mathematical section, I'm fine with the majority of reviewers thought it was an overkill. I will not be pushy on this :)

Reply:
@NicolasHug First excuse - it was late :blush: Second excuse, GLM literature is lazy and skips the conditional most of the time, e.g. E[Y] really means E[Y|X].

More seriously: Maybe, I missed that piece in the actual user guide, but I can't find a reference as to what regressors do actually estimate/predict (most of the time E[Y|X]). That's why I bring this up again (last time, promise). Should we add that info somewhere in the UG?

Reply:
> Should we add that info somewhere in the UG?

That's a possibility, yes. Though I would prefer doing that in another PR to make sure the UG is overall consistent, since this would be a non-trivial update.

In the mean time, I still think we can have a reasonable UG for the GLMs that doesn't mention E[Y/X], we can just talk about predictions.